R/pkgs/hospitalization/R/hospdeath.R

```diff
@@ -78,68 +251,321 @@ build_hospdeath_par <- function(p_hosp, p_death, p_ICU, p_vent, data_filename, s
-         .x$icu_curr <- cumsum(.x$incidICU) - lag(cumsum(.x$incidICU),n=ICU_dur_,default=0)
-         .x$geo_ind <- .y$geo_ind
-         return(.x)
-       }) %>%
-       do.call(what=rbind) %>%
-       arrange(date_inds, geo_ind)
-
-    outfile <- paste0(root_out_dir,'/', data_filename,'/',scenario_name,'-',s,'.csv')
-    outdir <- gsub('/[^/]*$','',outfile)
-    if(!dir.exists(outdir)){
-      dir.create(outdir,recursive=TRUE)
-    }
-    write.csv(res,outfile)
+             hosp_curr = 0)) %>%
+      # get sim nums
+      select(-geoid, -sim_num) %>%
+      separate(uid, c("geoid", "sim_num"), sep="-", remove=FALSE) %>%
+      mutate(date_inds = as.integer(time - min(time) + 1),
+             geo_ind = as.numeric(as.factor(geoid))) %>%
+      arrange(geo_ind, date_inds) %>%
+      group_by(geo_ind) %>%
+      group_map(function(.x,.y){
+        .x$hosp_curr <- cumsum(.x$incidH) - lag(cumsum(.x$incidH),
+                                                n=R_delay_,default=0)
+        .x$icu_curr <- cumsum(.x$incidICU) - lag(cumsum(.x$incidICU),
+                                                 n=ICU_dur_,default=0)
+        .x$vent_curr <- cumsum(.x$incidVent) - lag(cumsum(.x$incidVent),
+                                                   n=Vent_dur_)
+        .x$geo_ind <- .y$geo_ind
+        return(.x)
+      }) %>%
+      do.call(what=rbind) %>%
+      replace_na(
+        list(vent_curr = 0,
+             icu_curr = 0,
+             hosp_curr = 0)) %>%
+      arrange(date_inds, geo_ind)
+    write_hosp_output(root_out_dir, data_dir, dscenario_name, s, res, use_parquet)
+    NULL
+  }
+  doParallel::stopImplicitCluster()
+}
+
+
+##'
+##' Build a set of sampled hospitalizations, deaths, and recoveries
+##' from the incident infection data from the simulation model.
+##' Use GEOID-specific, age-adjust probabilities of hospitalization,
+##' death, and ICU admission
+##'
+##' @param prob_dat df of p_hosp, p_death, p_ICU, p_vent, GEOID
+##' @param scl_fac scaling factor for the probability symptomatic infection
+##' @param data_dir Path to the directory that contains the CSV output of the simulation model
+##' @param dscenario_name The name of the scenario we are analyzing here (e.g., "highdeath", "meddeath", etc.)
+##' @param time_hosp_pars parameters for time from onset to hospitalization distribution
+##' @param time_ICU_pars parameters for time from hospitalization to ICU
+##' @param time_vent_pars parameters for time from ICU to time on ventilator
+##' @param time_onset_death_pars parameters for time from onset to death distribution
+##' @param time_disch_pars parameters for time from hospitalization to discharge parameters
+##' @param time_ICUdur_pars parameetrs for time of ICU duration
+##' @param cores The number of CPU cores to run this model on in parallel
+##' @param root_out_dir Path to the directory to write the outputs of this analysis
+##'
+##' @export
+build_hospdeath_geoid_par <- function(
```

Comment:
are we using build_hospdeath_geoid_par anywhere (we shouldn't be but maybe @salauer knows something I don't)? I might leave out of master for time being

Reply:
I don't use it. I just kept it updated in case anyone else did.

Reply:
I've deleted this in dataseed_merge